client/my-sites/checkout/src/hooks/use-coupon-field-state.ts

```diff
@@ -35,25 +35,17 @@ export default function useCouponFieldState(
 	const handleCouponSubmit = useCallback( () => {
 		const trimmedValue = couponFieldValue.trim();
 
-		if ( isCouponValid( trimmedValue ) ) {
-			reduxDispatch(
-				recordTracksEvent( 'calypso_checkout_composite_coupon_add_submit', {
-					coupon: trimmedValue,
-				} )
-			);
-
-			applyCoupon( trimmedValue ).catch( () => {
-				// Nothing needs to be done here. CartMessages will display the error to the user.
-			} );
-
-			return;
-		}
-
 		reduxDispatch(
-			recordTracksEvent( 'calypso_checkout_composite_coupon_add_error', {
-				error_type: 'Invalid code',
+			recordTracksEvent( 'calypso_checkout_composite_coupon_add_submit', {
+				coupon: trimmedValue,
 			} )
 		);
+
+		applyCoupon( trimmedValue ).catch( () => {
+			// Nothing needs to be done here. CartMessages will display the error to the user.
```

Comment:
For context, here is why `catch` is still useful https://github.com/Automattic/wp-calypso/pull/66540